src/core/transaction.js

```diff
@@ -1037,26 +1027,14 @@ Transaction.prototype.rippleLineSet = function(options_) {
 };
 
 Transaction.prototype.setLimit = function(amount) {
-  // if (this.getType() !== 'TrustSet') {
-    // throw new Error('TransactionType must be TrustSet to use LimitAmount');
-  // }
-
   return this._setAmount('LimitAmount', amount, {no_native: true});
 };
 
 Transaction.prototype.setQualityIn = function(quality) {
-  // if (this.getType() !== 'TrustSet') {
-    // throw new Error('TransactionType must be TrustSet to use QualityIn');
-  // }
-
   return this._setUInt32('QualityIn', quality);
 };
 
 Transaction.prototype.setQualityOut = function(quality) {
-  // if (this.getType() !== 'TrustSet') {
```

Comment:
This is now handled by the "field unknown for type" like messages in SerializedObject.from_json ?